### PR TITLE
Convert TaskViewFactory to Kotlin

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
@gino-m PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/editsubmission/TaskViewFactory.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      )
+    binding.lifecycleOwner = fragment
+    binding.setVariable(BR.viewModel, viewModelFactory.create(getViewModelClass(taskType)))
+    assignGeneratedId(binding.root)
+    return binding
+  }
+
+  companion object {
+    fun getViewModelClass(taskType: Task.Type): Class<out AbstractTaskViewModel> {
+      return when (taskType) {
```
Comment (JSunde) [inline ground/src/main/java/com/google/android/ground/ui/editsubmission/TaskViewFactory.kt:59]: nit: could be single expression function

File: ground/src/main/java/com/google/android/ground/ui/editsubmission/TaskViewFactory.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        Task.Type.NUMBER -> NumberTaskViewModel::class.java
+        Task.Type.DATE -> DateTaskViewModel::class.java
+        Task.Type.TIME -> TimeTaskViewModel::class.java
+        else -> throw IllegalArgumentException("Unsupported task type: $taskType")
+      }
+    }
+
+    @LayoutRes
+    private fun getLayoutId(taskType: Task.Type): Int {
+      return when (taskType) {
```
Comment (JSunde) [inline ground/src/main/java/com/google/android/ground/ui/editsubmission/TaskViewFactory.kt:72]: ditto

**Review by JSunde — APPROVED** [review]

**Review by gino-m — APPROVED** [review]
LGTM, modulo resolving @JSunde's suggestions. Thanks for migrating!

File: ground/src/main/java/com/google/android/ground/ui/editsubmission/TaskViewFactory.kt (hunk 1 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/editsubmission/TaskViewFactory.kt]: Done

File: ground/src/main/java/com/google/android/ground/ui/editsubmission/TaskViewFactory.kt (hunk 2 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/editsubmission/TaskViewFactory.kt]: Done

**Pull request merged**